utils/stm32stack.pl

```diff
@@ -164,17 +164,48 @@
     $call_graph{$from} = \%resolved;
 }
 
+#
+# All Page entry functions are called from well-known functions
+#
+foreach (keys %call_graph) {
+	my $from = $_;
+	my $callees = $call_graph{$from};
+	my %resolved;
+
+	if ($from =~ /^PAGE_ChangeByID@/)
+	{
+		foreach (keys %call_graph) {
+			$call_graph{$from}->{$_} = 1 if /^PAGE_.+Init@/;
+			$call_graph{$from}->{$_} = 1 if /^PAGE_.+Exit@/;
+		}
+	}
+
+	if ($from =~ /^PAGE_Exit@/)
+	{
+		foreach (keys %call_graph) {
+			$call_graph{$from}->{$_} = 1 if /^PAGE_.+Exit@/;
+		}
+	}
+
+	if ($from =~ /^PAGE_Event@/)
+	{
+		foreach (keys %call_graph) {
+			$call_graph{$from}->{$_} = 1 if /^PAGE_.+Event@/;
+		}
+	}
+}
+
 # Create fake edges and nodes to account for dynamic behaviour.
```

Comment:
Please put this into its own function.  The code is already terribly organized, no reason to make it worse

Reply:
Can u help? I don't really understand how perl to pass parameter by ref or by val.

Reply:
I have cleaned it up and put it here:
https://gist.github.com/PhracturedBlue/3b3f0dd8825b85910249de726d99e326

Seems to work ok for me